### PR TITLE
Document tunnel option for development

### PR DESCRIPTION
**Pull request by breyed**
Using a tunnel to set up development can be quicker and easier than hostname and certification modification if you don't need to run spend a lot of time on authentication.

## Checklist 🧢

- [x] Documentation
~[ ] Tests~
- [x] Ready to be merged

### CODE REVIEW

File: docs/docs/providers/apple.md [hunk 1]
```diff
@@ -57,9 +57,20 @@ Apple requires all sites to run HTTPS (including local development instances).
 Apple doesn't allow you to use localhost in domains or subdomains.
 :::
 
-### Host name resolution
+### Using a tunnel
 
-Edit your host file and point your site to `127.0.0.1`.
+One developemnt option to avoid using HTTP or localhost is to access your site through a tunnel such as [ngrok](https://ngrok.com/).
```
Comment (ndom91) [inline docs/docs/providers/apple.md:62]:
```suggestion
You can use a tunneling service like [ngrok](https://ngrok.com/) to simplify local development and easily get an HTTPS URL required for Apple OAuth settings.
```

File: docs/docs/providers/apple.md [hunk 2]
```diff
@@ -57,9 +57,20 @@ Apple requires all sites to run HTTPS (including local development instances).
...
+
+1. Follow the instructions to set up the tunnel and install the agent.
```
Comment (ndom91) [inline docs/docs/providers/apple.md:64]:
```suggestion
1. Follow the [instructions](https://ngrok.com/download) to set up an ngrok tunnel.
```

File: docs/docs/providers/apple.md [hunk 3]
```diff
@@ -57,9 +57,20 @@ Apple requires all sites to run HTTPS (including local development instances).
...
+2. In `.env.local`, temporarily set `NEXTAUTH_URL` to be your assigned tunnel URL, something like `https://12345.ngrok.io`.
+3. Temporarily add the tunnel domain (`12345.ngrok.io`) and reply URL (`https://12345.ngrok.io/api/auth/callback/apple`) into your sign-in service identifier in [Certificates, Identifiers & Profiles](https://developer.apple.com/account/resources/identifiers/serviceId/edit/). 
+4. Browse to your tunnel URL and test sign in.
+
+Once you've tested authentication and created a local account, you can remove the tunnel references and switch back to using localhost.
+
+### Modifying your local host name
+
+An altnerative approach is to make your development computer see itself as a non-local website. Edit your host file and point your site to `127.0.0.1`.
```
Comment (ndom91) [inline docs/docs/providers/apple.md:73]:
```suggestion
An alternative approach is to setup a hostname and SSL certificates for your local computer. 

Begin by editing your host file (`/etc/hosts`) and point your site to `127.0.0.1`. For example, by adding the line: `127.0.0.1 mysite.com`
```

**Review by ndom91 — CHANGES_REQUESTED** [review]
Some small wording feedback, but generally I like the addition. Using ngrok, or a similar tunneling service, would make setting up Apple OAuth locally much easier. Thanks!

**Comment by breyed** [comment]
After creating this PR, I discovered vercel/next.js#10935 which covers dives deeper into the HTTPS options. In summary:
* [Custom server](https://github.com/vercel/next.js/discussions/10935#discussioncomment-584332) - A variation of what's in NextAuth Apple docs now. A related note: For a newcomer, the existing NextAuth docs can be confusing and cause errors (it was for me, anyway). For the line to add to /etc/hosts, the sudo command resulted in a permission denied error (I'm using macOS). (It was easy to work around with `sudo vi`.) More importantly, what's happening with the custom server and its implications isn't clear to the uninitiated.
* [Reverse proxy](https://github.com/vercel/next.js/discussions/10935#discussioncomment-2236301) - This option seems easy but comes with deprecationa and security warnings.

It's worth considering all the options in crafting curated guidance for the NextAuth docs.

**Comment by ndom91** [comment]
If you want to develop your next.js app locally with HTTPS I highly recommend you go the reverse proxy route. i.e. generate a cert locally and set it up with nginx or caddy or something similar. This however obviously involves a bit of sys admin type work and if you're not familiar with it, your recommended solution of simply firing up an ngrok tunnel is much much easier.

Working with a custom next.js server and adding https just for local development isn't worth the headache imo. Just generate a cert, add the file paths to an nginx config and point that nginx host to `localhost:3000`, or wherever your normal dev URL is.

I think overall developing with HTTPS is a rather niche requirement. The only thing I can think of really requiring this is something like what you documented - if an OAuth provider or other 3rd party service requires it, or some of the newer web API's which only work over HTTPS like web bluetooth, geolocation, etc. none of which are related to the core NextAuth.js feature's afaik.

**Pull request closed without merging**